src/visualization/makie_visualization.jl

```diff
@@ -0,0 +1,348 @@
+  flat_wire_points = Makie.@lift get_flat_points($wire_points, $z_offset)
+  wire_mesh_flat = Makie.lines!(ax, flat_wire_points, color=:black)
+
+  # reset colorbar each time solution changes
+  Makie.Colorbar(fig[1, 3], limits = Makie.@lift(extrema($solution_z)), colormap = :viridis, flipaxis = false)
+
+  # syncs the toggle to the mesh
+  Makie.connect!(wire_mesh_top.visible, toggle_solution_mesh.active)
+  Makie.connect!(wire_mesh_bottom.visible, toggle_solution_mesh.active)
+  Makie.connect!(wire_mesh_flat.visible, toggle_mesh.active)
+
+  # On OSX, shift-command-4 for screenshots triggers a constant "up-zoom".
+  # To avoid this, we remap up-zoom to the right shift button instead.
+  Makie.cameracontrols(ax.scene).attributes[:up_key][] = Makie.Keyboard.right_shift
+
+  # typing this pulls up the figure (similar to display(plot!()) in Plots.jl)
+  fig
+end
+
+# ================== new Makie plot recipes ====================
+
+@Makie.recipe(TrixiHeatmap, plot_data_series) do scene
+  Makie.Theme(;)
+end
+
+function Makie.plot!(myplot::TrixiHeatmap)
+  pds = myplot[:plot_data_series][]
+
+  plotting_mesh = generate_plotting_triangulation(pds; set_z_coordinate_zero = true)
+
+  @unpack variable_id = pds
+  pd = pds.plot_data
+  solution_z = vec(StructArrays.component(pd.u, variable_id))
+  Makie.mesh!(myplot, plotting_mesh, color=solution_z, shading=false)
+
+  # Makie hides keyword arguments within `myplot`; see also
+  # https://github.com/JuliaPlots/Makie.jl/issues/837#issuecomment-845985070
+  plot_mesh = if haskey(myplot, :plot_mesh)
+    myplot.plot_mesh[]
+  else
+    true # default to plotting the mesh
+  end
+
+  if plot_mesh==true
+    xyz_wireframe = generate_plotting_wireframe(pds; set_z_coordinate_zero = true)
+    Makie.lines!(myplot, xyz_wireframe, color=:lightgrey)
+  end
+
+  myplot
+end
+
+# redirects Makie.plot(pd::PlotDataSeries2D) to custom recipe TrixiHeatmap(pd)
+Makie.plottype(::Trixi.PlotDataSeries2D{<:Trixi.UnstructuredPlotData2D}) = TrixiHeatmap
+
+# Makie does not yet support layouts in its plot recipes, so we overload `Makie.plot` directly.
+function Makie.plot(sol::TrixiODESolution;
+                    plot_mesh=true, solution_variables=nothing)
+  Makie.plot(PlotData2D(sol, solution_variables=solution_variables); plot_mesh=plot_mesh)
+end
+
+function Makie.plot(pd::UnstructuredPlotData2D, fig = Makie.Figure();
+                    plot_mesh = true)
+  Makie.plot!(fig, pd; plot_mesh=plot_mesh)
+  fig
+end
+
+function Makie.plot!(fig, pd::UnstructuredPlotData2D;
+                     plot_mesh = true)
+  # Create layout that is as square as possible, when there are more than 3 subplots.
+  # This is done with a preference for more columns than rows if not.
+  if length(pd) <= 3
+    cols = length(pd)
+    rows = 1
+  else
+    cols = ceil(Int, sqrt(length(pd)))
+    rows = ceil(Int, length(pd)/cols)
+  end
+
```

Comment:
Would it be possible to add the axis labels by default as we do for Plots.jl?

Reply:
What do you mean?

Reply:
But there don't seem to be axis labels on the Plots recipes, at least on the doc images https://trixi-framework.github.io/Trixi.jl/dev/visualization/

Reply:
Yeah, that figure is old and we should update it (#788). In newer versions, we use axis labels, see https://github.com/trixi-framework/Trixi.jl/blob/main/src/visualization/plot_recipes.jl#L309-L310 and https://github.com/trixi-framework/Trixi.jl/blob/main/src/visualization/plot_recipes.jl#L281-L291

Reply:
Tracked in #792.